src/database/helpers.rs

```diff
@@ -68,21 +68,31 @@ pub async fn delete_name(
 }
 
 #[instrument(skip(postgres))]
-pub async fn update_name(
+pub async fn update_name_attributes(
     name: String,
     attributes: HashMap<String, String>,
     postgres: &PgPool,
-) -> Result<sqlx::postgres::PgQueryResult, sqlx::error::Error> {
-    let insert_name_query = "
+) -> Result<HashMap<String, String>, sqlx::error::Error> {
+    let update_attributes_query = "
       UPDATE names SET attributes = $2::hstore, updated_at = NOW()
-        WHERE name = $1
+        WHERE name = $1 
+        RETURNING attributes::json
     ";
-    sqlx::query::<Postgres>(insert_name_query)
+    let result: (serde_json::Value,) = sqlx::query_as(update_attributes_query)
```

Comment:
Shouldn't this work?

```suggestion
    let result: serde_json::Value = sqlx::query_as(update_attributes_query)
```

Reply:
No, complaining about the trait. This code smells to me as well, so I [rewrote it](https://github.com/WalletConnect/blockchain-api/pull/509/commits/34bf8bbcd37bdd5cf976cdf963ecf7094b60acae). Thanks!